src/bin/rad-ci.rs

```diff
@@ -0,0 +1,161 @@
+use std::convert::TryInto;
+use std::io::{Read, Write};
+use std::net::TcpStream;
+
+use rad_ci::{Options, HELP};
+use rad_terminal::components as term;
+
+use anyhow::Context;
+use ssh2::Session;
+
+fn main() {
+    rad_terminal::args::run_command::<Options, _>(HELP, "CI", run);
+}
+
+fn run(options: Options) -> anyhow::Result<()> {
+    let seed = options.seed.context("Seed is invalid")?;
+    // SSH into server
+    let tcp = TcpStream::connect(format!("{}:22", &seed))?;
+    let mut sess = Session::new()?;
+    sess.set_tcp_stream(tcp);
+    sess.handshake()?;
+
+    // Authenticate
+    let spinner = term::spinner(&format!("SSH into remote {}", &seed));
+    if options.ssh_index == 0 {
+        sess.userauth_agent(&options.ssh_user)?;
+    } else {
+        let mut agent = sess.agent()?;
+        agent.connect()?;
+        agent.list_identities()?;
+        let identities = agent.identities()?;
+        let identity = identities.get(options.ssh_index).context(format!(
+            "Couldn't retrieve ssh-agent key #{}",
+            options.ssh_index
+        ))?;
+        agent.userauth(&options.ssh_user, identity)?;
```

Comment:
Passing a key index is kind of weird, can we not pass a key path?

Reply:
(I think avoiding key path in favor of `ssh-agent` is a better option, now that I used it)

Reply:
I went ahead and removed index, instead now we iterate over all keys.